src/Analyzer/Pass/Expression/Casts.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace PHPSA\Analyzer\Pass\Expression;
+
+use PhpParser\Node\Expr;
+use PHPSA\Analyzer\Pass\AnalyzerPassInterface;
+use PHPSA\Context;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use PHPSA\CompiledExpression;
+
+class Casts implements AnalyzerPassInterface
+{
+    /**
+     * @param Expr $expr
+     * @param Context $context
+     * @return bool
+     */
+    public function pass(Expr $expr, Context $context)
+    {
+
+        $castType = CompiledExpression::UNKNOWN;
+
+        switch (true) {
```

Comment:
I suggest to use `switch(get_class($expr))`

And `case Expr\Cast\Array_::class` it's more easy to read and can up code quality

Thanks